internals/auth/signature.go

```diff
@@ -115,22 +130,23 @@ func (c signer) AddAuthentication(r *http.Request) error {
 		return errio.Error(err)
 	}
 
-	signature, err := c.key.Sign(message)
+	signature, err := s.signer.Sign(message)
 	if err != nil {
 		return errio.Error(err)
 	}
 
 	base64EncodedSignature := base64.StdEncoding.EncodeToString(signature)
 
-	fingerprint, err := c.key.Public().Fingerprint()
+	id, err := s.signer.ID()
 	if err != nil {
 		return errio.Error(err)
 	}
 
 	r.Header.Set("Authorization",
-		fmt.Sprintf("%s %s:%s",
-			MethodTagSignature,
-			fingerprint,
+		fmt.Sprintf("%s %s:%s:%s",
```

Comment:
Looking into the [Mozilla Developer Page](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization), it seems that the standard is to incorporate everything about the type into the first element of the header. So that would make something like:
```
SH1-HMAC-SHA256 <id>:<signature>
```

What do you think about this?

Reply:
Sounds good to me. The types shouldn't vary too much to require many differentiators. A version tag would be nice though.

Reply:
I've changed it to use `SH1-<signing method> <id>:<signature>`

Reply:
Ah, so `SH1` is `SecretHub v1`, got it. At a glance I thought it had something to do with `SHA`

Reply:
Hmm. That can indeed be confusing. We don't want to let people think we're using SHA1.Should we use `SecretHub-v1-` or `SECRETHUB1-` instead?